mutters-core/src/main/java/com/rabidgremlin/mutters/core/bot/IntentBotResponse.java

```diff
@@ -0,0 +1,63 @@
+/* Licensed under Apache-2.0 */
+package com.rabidgremlin.mutters.core.bot;
+
+import java.util.List;
+
+import com.rabidgremlin.mutters.core.MatcherScores;
+
+/**
+ * A response from an IntentBot. Extends BotResponse to include the name of the
+ * matched intent and the scores of the matching.
+ * 
+ * @author rabidgremlin
+ *
+ */
+public class IntentBotResponse extends BotResponse
+{
+  /** The name of the matched intent. */
+  private String matchedIntent;
+
+  /** The scores generated by the intent matcher. */
+  private MatcherScores matchingScores;
+
+  /**
+   * Constructor
+   * 
+   * @param response       The response text.
+   * @param hint           Any hint text for the client to display.
+   * @param askResponse    True if this the bot is expecting a further response
+   *                       from the user.
+   * @param attachments    The attachments for the response.
+   * @param quickReplies   List of suggested replies that the user could use.
+   * @param matchedIntent  The name of the matched intent.
+   * @param matchingScores The scores generated by the intent matching.
+   */
+  public IntentBotResponse(String response, String hint, boolean askResponse, List<BotResponseAttachment> attachments,
+      List<String> quickReplies, String matchedIntent, MatcherScores matchingScores)
```

Comment:
thoughts on making the ```matchedIntent``` an ```Intent``` instead of ```String``` ?

Reply:
mmm, Can do. Will expose more of the "internals" in the response but I don't see the harm.

Reply:
Updated.